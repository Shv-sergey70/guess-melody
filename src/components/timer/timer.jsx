import React from 'react';
import PropTypes from 'prop-types';

const Timer = ({minutes, seconds, isFinishing}) => {
  return (
    <div className={`timer__value ${isFinishing ? `timer__value--finished` : ``}`} xmlns="http://www.w3.org/1999/xhtml">
      <span className="timer__mins">{minutes}</span>
      <span className="timer__dots">:</span>
      <span className="timer__secs">{seconds}</span>
    </div>
  );
};

Timer.propTypes = {
  minutes: PropTypes.string.isRequired,
  seconds: PropTypes.string.isRequired,
  isFinishing: PropTypes.bool.isRequired
};

export default Timer;
