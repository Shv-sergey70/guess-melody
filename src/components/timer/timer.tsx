import * as React from 'react';

type Props = {
  minutes: string,
  seconds: string,
  isFinishing: boolean
};

const Timer: React.FunctionComponent<Props> = ({minutes, seconds, isFinishing}) => {
  return (
    <div className={`timer__value ${isFinishing ? `timer__value--finished` : ``}`}>
      <span className="timer__mins">{minutes}</span>
      <span className="timer__dots">:</span>
      <span className="timer__secs">{seconds}</span>
    </div>
  );
};

export default Timer;
