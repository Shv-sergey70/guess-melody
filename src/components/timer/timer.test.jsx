import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Timer} from "./timer";

configure({adapter: new Adapter()});

test(`Timer renders correctly`, () => {
  const tickMock = jest.fn();

  const tree = shallow(
      <Timer
        time={119}
        tick={tickMock} />
  );

  expect(tree).toMatchSnapshot();
});
