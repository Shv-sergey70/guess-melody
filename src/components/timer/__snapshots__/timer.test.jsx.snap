// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer renders correctly Timer has a lot of time - usual behaviour 1`] = `
<div
  className="timer__value "
  xmlns="http://www.w3.org/1999/xhtml"
>
  <span
    className="timer__mins"
  >
    02
  </span>
  <span
    className="timer__dots"
  >
    :
  </span>
  <span
    className="timer__secs"
  >
    52
  </span>
</div>
`;

exports[`Timer renders correctly Timer has low time - should be added finishing class 1`] = `
<div
  className="timer__value timer__value--finished"
  xmlns="http://www.w3.org/1999/xhtml"
>
  <span
    className="timer__mins"
  >
    00
  </span>
  <span
    className="timer__dots"
  >
    :
  </span>
  <span
    className="timer__secs"
  >
    18
  </span>
</div>
`;
